### PR TITLE
Add an option to `GlueJobOperator` to stop the job run when the TI is killed

### LINKED ISSUE #32137: AWS - GlueJobOperator - stop job on task stop
### Description

if you stop a running GlueJobOperator task ( mark fail ) the operator do not call aws glue to stop the runnning task

so the airflow task stop but the aws glue job continue to run

### Use case/motivation

Like the KPO would be nice to stop the aws glue job in case the operator is stop

currently the logs of operator are

```log
[2023-06-26, 10:07:32 UTC] {glue.py:149} INFO - Initializing AWS Glue Job: XXXXXXXX. Wait for completion: True
[2023-06-26, 10:07:32 UTC] {glue.py:290} INFO - Checking if job already exists: XXXXXXXXXXX
[2023-06-26, 10:07:32 UTC] {base.py:73} INFO - Using connection ID 'XXXXXXXX' for task execution.
[2023-06-26, 10:07:32 UTC] {credentials.py:1051} INFO - Found credentials from IAM Role: XXXXXXXXXXXX
[2023-06-26, 10:07:33 UTC] {glue.py:169} INFO - You can monitor this Glue Job run at: https://console.aws.amazon.com/gluestudio/home?region=eu-west-1#/job/test_ip/run/XXXXXXXXXXXXXXX
[2023-06-26, 10:07:33 UTC] {glue.py:265} INFO - Polling for AWS Glue Job test_ip current run state with status RUNNING
[2023-06-26, 10:07:39 UTC] {glue.py:265} INFO - Polling for AWS Glue Job test_ip current run state with status RUNNING
[2023-06-26, 10:07:45 UTC] {glue.py:265} INFO - Polling for AWS Glue Job test_ip current run state with status RUNNING
[2023-06-26, 10:07:51 UTC] {glue.py:265} INFO - Polling for AWS Glue Job test_ip current run state with status RUNNING
[2023-06-26, 10:07:57 UTC] {glue.py:265} INFO - Polling for AWS Glue Job test_ip current run state with status RUNNING
[2023-06-26, 10:08:02 UTC] {local_task_job.py:276} WARNING - State of this instance has been externally set to failed. Terminating instance.
[2023-06-26, 10:08:02 UTC] {process_utils.py:129} INFO - Sending Signals.SIGTERM to group 13175. PIDs of all processes in the group: [13175]
[2023-06-26, 10:08:02 UTC] {process_utils.py:84} INFO - Sending the signal Signals.SIGTERM to group 13175
[2023-06-26, 10:08:02 UTC] {taskinstance.py:1488} ERROR - Received SIGTERM. Terminating subprocesses.
[2023-06-26, 10:08:02 UTC] {process_utils.py:79} INFO - Process psutil.Process(pid=13175, status='terminated', exitcode=0, started='10:07:31') (13175) terminated with exit code 0


```

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #32137

This PR adds a new boolean parameter `stop_job_run_on_kill`  to `GlueJobOperator`, when it is set to `True`, the Glue job run will be stopped if the TI is killed (because of a timeout or a SIGTERM/clear).

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/glue.py [hunk 1]
```diff
@@ -123,12 +125,11 @@ def __init__(
...
+        self.stop_job_run_on_kill = stop_job_run_on_kill
+        self._job_run_id: str | None = None
 
-    def execute(self, context: Context):
-        """Execute AWS Glue Job from Airflow.
-
-        :return: the current Glue job ID.
-        """
+    @cached_property
+    def glue_job(self) -> GlueJobHook:
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/operators/glue.py]:
```suggestion
    def glue_job_hook(self) -> GlueJobHook:
```

**Review by vincbeck — CHANGES_REQUESTED** [review]
A simple nit. I think `glue_job` can be confusing and can make you think you return the actual glue job

File: airflow/providers/amazon/aws/operators/glue.py [hunk 2]
```diff
@@ -100,6 +101,7 @@ def __init__(
         verbose: bool = False,
         update_config: bool = False,
         job_poll_interval: int | float = 6,
+        stop_job_run_on_kill: bool = False,
```
Comment (o-nikolas) [inline airflow/providers/amazon/aws/operators/glue.py:105]: Can you add a `:param stop_job_run_on_kill: ...` description to the method doc string please?

**Comment by hussein-awala** [comment]
@vincbeck @o-nikolas Thank you for your suggestion. Could you please take a second look?

**Review by o-nikolas — APPROVED** [review]

**Review by vincbeck — APPROVED** [review]

**Pull request merged**